### DISCUSSION TITLE
Need to import text from text file to form textarea field or state

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#14303] [Help]
Hi all,

I am trying to implement with JS using embed element or a form button a way to read a text file contents on user HDD and place contents directly in a textfield form element or at state to place then at the form field.

First I tried to create an Input file button in embed and use the Filereader API . In that case I can select the file and read the file contents to an HTML element created also in embed by I cannot get any page form elements to alter its contents. 
The HTML code that worked for embed is following:
```
<p id="outputfile"></p>
<input type="file" name="inputfile" id="inputfile" accept=".csv, application/vnd.openxmlformats-officedocument.spreadsheetml.sheet, application/vnd.ms-excel" 
onchange="(event => {
    const input = event.target;
    const file = input.files[0];
    if (file) {
        const reader = new FileReader();
        reader.onload = event => document.getElementById('outputfile').innerText = event.target.result;
        reader.onerror = () => document.getElementById('outputfile').innerText = 'Error reading file';
        reader.readAsText(file);
    } else {
        document.getElementById('outputfile').innerText = 'No file selected';
    }
})(event)">
```
In 2nd case I tried to create a Button setting a state calling an async function calling the window.showOpenFilePicker() function of the File System Access API with ultimate purpose to get file contents to , but code cannot be executed because function window.showOpenFilePicker()  is not recognized though I am on Chrome browser and this method should work as proven by this example page: https://googlechromelabs.github.io/text-editor/

Could you help me make any of these options work or suggest another option to get file contents inside a form field from Budibase app frontend?

**Comment by gmentzik** [comment]
I am posting here a solution I found after reading the wonderful explanation provided at discussion [How to Add Custom JavaScript](https://github.com/Budibase/budibase/discussions/6839) by @R2bEEaton.

With i can directly import the text contents of a file to a textarea field. Needs some improvements but it works for now. 

```
// EMBED text code test 5
//  Creates input file button. Add onchange handling on input file button. Modifies Textarea text with text file contents.
<div id="FileSelectGbI"></div>
<img style="display: none;" src onerror="
   var ftas = document.getElementsByTagName('textarea');
   let butContainter = document.getElementById('FileSelectGbI');
   // alert('Hello world! Found: ' + ftas.length);
   var textarea0 = ftas[0];
   // textarea0.value = 'Modified text by value';
   
   console.log(ftas[0]);

   var inputel = document.createElement('input');
   inputel.type = 'file';
   // inputel.textContent  = 'Select GI File';  // HTML is not updating but shows in console
   
   butContainter.append(inputel);

   inputel.onchange = function(event) {
   // Get the selected file(s)
   var files = event.target.files;

   // Do something with the selected file(s)
   if (files.length > 0) {
	 const file = files[0];
	 console.log('Selected file:', file.name);
	 // Add additional actions here, e.g., uploading the file or displaying its content
	 const reader = new FileReader();
	 reader.onload = event => textarea0.value = event.target.result;
	 reader.onerror = () => textarea0.value = 'Error reading file';
	 reader.readAsText(file);
   } else {
	 textarea0.value = 'No file selected';
   }
	
  };
">
```

I think you still need to click in and out of textarea though so Budibase can understand the changed text.

**Reply by R2bEEaton** [reply]
Excellent, glad you were able to figure it out!
